### PR TITLE
update supertest dependency

### PR DESCRIPTION
**Pull request by nabdelgadir**
### Description
Updated the `supertest` dependency and modified a couple tests to cover the changes. For the `expect err to have property code` test, I submitted a PR to `supertest` to add that property to the error (hasn't been reviewed yet), so I kept the old test in if they accept the change and we can revert back to the old test.

#### Related issues

- connect to #498 

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: test/end-to-end.js [hunk 1]
```diff
@@ -1035,7 +1035,7 @@ describe('end-to-end', function() {
       });
     });
 
-    it('stops the app started by the workspace', function(done) {
+    it.only('stops the app started by the workspace', function(done) {
```
Comment (jannyHou) [inline test/end-to-end.js]: no only :)

Comment (nabdelgadir) [reply test/end-to-end.js]: Oh, whoops, thanks!

**Comment by jannyHou** [comment]
BTW the commit message has formatting problems, could you check the error and rephrase accordingly?

**Comment by nabdelgadir** [comment]
> BTW the commit message has formatting problems, could you check the error and rephrase accordingly?

For both the commit linter and PR linter, it says `stderr: git: 'lfs' is not a git command.` when it tries to run `git lfs pull origin`, not sure why it's attempting to run that as it didn't on the last PR.

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
@jannyHou @nabdelgadir I missed a host when I was installing `git-lfs` on our CI machines. It's working now.

**Comment by nabdelgadir** [comment]
@rmg thank you!

File: test/end-to-end.js [hunk 2]
```diff
@@ -1044,7 +1044,8 @@ describe('end-to-end', function() {
             if (err) return done(err);
             request(appUrl).get('/api/products')
               .end(function(err) {
-                expect(err).to.have.property('code', 'ECONNREFUSED');
+                expect(err.message, 'err.message').to.contain('ECONNREFUSED');
+                // expect(err).to.have.property('code', 'ECONNREFUSED');
```
Comment (dhmlau) [inline test/end-to-end.js]: remove this commented line?

**Comment by nabdelgadir** [comment]
@dhmlau @jannyHou put a comment explaining why I left the comment there. :slightly_smiling_face:

**Pull request merged**